git/git.go

```diff
@@ -37,6 +39,31 @@ func New(directory string) (*Git, error) {
 
 }
 
+func (g *Git) CommitID() (string, error) {
+	cmd := exec.Command(g.GitPath, "rev-parse", "--short", "HEAD")
+	cmd.Dir = g.Directory
+
+	log.Infof("$ git %s", strings.Join(cmd.Args, " "))
+
+	raw, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", err
+	}
+
+	id := strings.ToLower(strings.TrimSpace(string(raw)))
+
+	matched, err := regexp.MatchString("[0-9a-f]{7}", id)
+	if err != nil {
+		return "", err
```

Comment:
A string cannot be `nil`. I would have to set the type to `*string`, which isn't very neat.